Lib/test/test_shutil.py

```diff
@@ -33,6 +33,25 @@
 except ImportError:
     UID_GID_SUPPORT = False
 
+
+def _has_functional_lchmod():
+    if not support.can_symlink() or not hasattr(os, 'lchmod'):
+        return False
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        test_symlink = os.path.join(tmpdir, 'test')
+        os.symlink(os.devnull, test_symlink)
+        try:
+            os.lchmod(test_symlink, 0o777)
+        except OSError:
```

Comment:
I would prefer to see a check on the error code, or pass-through other exceptions.

Reply:
done